keys-component.js

```diff
@@ -59,7 +66,11 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
     self.modals.key = null;
   };
 
-  self.init = function(identity) {
+  self.onKeyGenerate = function(key) {
```

Comment:
This is returning a promise to the generate key modal.  The modal closes properly about 50% of the time.  Other times it closes after 3-5 seconds.  Seems like at $scope.$apply() issue, but that looks like it should be handled properly in the generate key modal.